src/openrct2/ride/VehiclePaint.cpp

```diff
@@ -962,17 +962,15 @@ static void PaintVehicleRiders(
 
 // 6D5214
 static void vehicle_sprite_paint(
-    paint_session& session, const Vehicle* vehicle, int32_t ebx, int32_t ecx, int32_t z,
+    paint_session& session, const Vehicle* vehicle, int32_t spriteNum, const vehicle_boundbox& bb, int32_t z,
```

Comment:
I see you have called ebx spriteNum. Most paint code would call this the `imageId`.
The way it mostly works (vehicle i know is special so perhaps we need a new term):

Object contains a `BaseImageId`. This is the index to the first sprite of all of the sprites that an object uses. It is assigned during loading.

The Paint function combines rotation and any other properties with `BaseImageId` to create an `uint32_t imageId`.
This is then combined with colours to create an `ImageId imageId`

The `ImageId imageId` is then used by the renderer to draw. It may modify the `uint32_t imageId` if needed for zoom sprites.

There is currently a lot of repeated use of `imageId` to mean a few different things but that is just how it works atm.

This function for example uses `baseImageId` where it really shouldn't its just building up a `ImageId imageId`

Reply:
Renaming spriteNum to imageId creates a collision between int32_t imageId and ImageId imageId. Renaming all the other functions are simpler.

Reply:
yup i know. I'm not really helping much here as i've no better naming convention. Underscores wouldn't look great we try limit that to class function variable name collisions. If you can't think of anything better than `spriteNum` leave as is.

Reply:
Ok, I will leave them all as spriteNum for consistency between all these functions.